stream-transformation-tool-service/src/main/java/uk/gov/justice/tools/eventsourcing/transformation/service/EventStreamTransformationService.java

```diff
@@ -104,15 +129,39 @@ private JsonEnvelope clearEventVersion(final JsonEnvelope event) {
         ).flatMap(identity());
     }
 
-    private boolean requiresTransformation(final Stream<JsonEnvelope> eventStream) {
-        return eventStream.filter(this::checkTransformations).count() > 0;
+    private TransformAction requiresTransformation(final Stream<JsonEnvelope> eventStream, UUID streamId) {
+
+        List<TransformAction> eventTransformationList = eventStream.map(t -> checkTransformations(t))
+                .flatMap(List::stream)
+                .distinct()
+                .collect(Collectors.toList());
+
+        if (eventTransformationList.size() == 0) {
+            return noAction(eventStream, streamId, "Stream {} did not require transformation stream ");
+        }
+        if (eventTransformationList.size() > 1) {
+            return noAction(eventStream, streamId, "Stream {} can not have multiple actions ");
```

Comment:
would it be useful to log/return what the multiple actions were?

Reply:
Done. added in logger the multiple action names